packages/flutter/test/material/text_field_test.dart

```diff
@@ -1612,19 +1611,83 @@ void main() {
       );
     }
 
-    // Explicitly allow selection on obscured text
-    await tester.pumpWidget(buildFrame(true, true));
+    // Explicitly disabled selection on obscured text.
+    await tester.pumpWidget(buildFrame(true, false));
     await tester.enterText(find.byType(TextField), 'abcdefghi');
     await skipPastScrollingAnimation(tester);
     expect(controller.selection.isCollapsed, true);
 
-    // Long press does select text
+    // Long press does select text.
     final Offset ePos2 = textOffsetToPosition(tester, 1);
     final TestGesture gesture2 = await tester.startGesture(ePos2, pointer: 7);
     await tester.pump(const Duration(seconds: 2));
     await gesture2.up();
     await tester.pump();
-    expect(controller.selection.isCollapsed, false);
+    expect(controller.selection.isCollapsed, true);
+  });
+
+  testWidgets('An obscured TextField is selected as one word', (WidgetTester tester) async {
+    final TextEditingController controller = TextEditingController();
+
+    await tester.pumpWidget(overlay(
+      child: TextField(
+        controller: controller,
+        obscureText: true,
+      ),
+    ));
+    await tester.enterText(find.byType(TextField), 'abcde fghi');
+    await skipPastScrollingAnimation(tester);
+
+    // Long press does select text.
+    final Offset bPos = textOffsetToPosition(tester, 1);
+    final TestGesture gesture = await tester.startGesture(bPos, pointer: 7);
+    await tester.pump(const Duration(seconds: 2));
+    await gesture.up();
+    await tester.pump();
+    final TextSelection selection = controller.selection;
+    expect(selection.isCollapsed, false);
+    expect(selection.baseOffset, 0);
+    expect(selection.extentOffset, 10);
+  });
+
+  testWidgets('An obscured TextField has correct default context menu', (WidgetTester tester) async {
+    final TextEditingController controller = TextEditingController();
+
+    await tester.pumpWidget(overlay(
+      child: TextField(
+        controller: controller,
+        obscureText: true,
+      ),
+    ));
+    await tester.enterText(find.byType(TextField), 'abcde fghi');
+    await skipPastScrollingAnimation(tester);
+
+    // Long press to select text.
+    final Offset bPos = textOffsetToPosition(tester, 1);
+    final TestGesture gesture = await tester.startGesture(bPos, pointer: 7);
```

Comment:
Can you just use longPressAt (https://master-api.flutter.dev/flutter/flutter_test/WidgetController/longPressAt.html) here and elsewhere where you needed a long press?